src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -140,13 +126,43 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
     relations.head
   }
 
+  private def hasParquetAsSourceFormatProperty(
+      relation: FileBasedRelation): Option[(String, String)] = {
+    if (relation.hasParquetAsSourceFormat) {
+      Some(IndexConstants.HAS_PARQUET_AS_SOURCE_FORMAT_PROPERTY -> "true")
+    } else {
+      None
+    }
+  }
+
+  private def hasLineageProperty(spark: SparkSession): Option[(String, String)] = {
+    if (hasLineage(spark)) {
+      Some(IndexConstants.LINEAGE_PROPERTY -> "true")
+    } else {
+      None
+    }
+  }
+
+  private def usesNestedFieldsProperty(indexConfig: IndexConfig): Option[(String, String)] = {
```

Comment:
`RefreshAtionBase` does NOT call `getIndexLogEntry`, however `RefreshActionBase` extends `CreateActionBase`. 

The `RefreshAction` is the one calling `getIndexLogEntry` from `CreateActionBase`.

Sorry, but I cannot understand what you're requesting with this comment. Please give more information.

Reply:
In RefreshAction.scala
```
override def logEntry: LogEntry = getIndexLogEntry(spark, df, indexConfig, indexDataPath)
```
The index config is escaped one, right? 
So need to override `usesNestedFieldsProperty` for RefreshActionBase as RefreshAction extends RefreshActionBase . . ?

Reply:
From what I know `IndexLogEntry` is what the user enters when it creates the index. When an index is created the given column names are NOT escaped. For example:

```
hs.createIndex(
  nestedDataset, 
  IndexConfig(
    "idx_nested", 
    indexedColumns = Seq("nested.nst.field1"), 
    includedColumns = Seq("id", "name", "nested.nst.field2")))
```

Reply:
in CreateActionBase.scala
```scala
    IndexLogEntry(
          indexConfig.indexName,
          CoveringIndex(
            CoveringIndex.Properties(
              CoveringIndex.Properties
                .Columns(resolvedIndexedColumns, resolvedIncludedColumns), <= 
              IndexLogEntry.schemaString(indexDataFrame.schema),
              numBuckets,
              coveringIndexProperties)),
          Content.fromDirectory(absolutePath, fileIdTracker),
          Source(SparkPlan(sourcePlanProperties)),
          Map())
```

We stored the indexed / included columns as resolved names.

So when refreshIndex, indexConfig.indexedColumn / includedColumn will be resolved one.. 
Could you check the parameter after refresh? It would be good to add the assert after some refresh index tests you created.